sample.py

```diff
@@ -0,0 +1,60 @@
+# This function adds two numbers
+def add(x, y):
+    return x + y
+
+
+# This function subtracts two numbers
+def subtract(x, y):
+    return x - y
+
+
+# This function multiplies two numbers
+def multiply(x, y):
+    return x * y
+
+
+# This function divides two numbers
+def divide(x, y):
+    if y == 0:
+        raise ValueError("Cannot divide by zero")
+    return x / y
+
+
+print("Select operation.")
+print("1.Add")
+print("2.Subtract")
+print("3.Multiply")
+print("4.Divide")
+
+while True:
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding global exception handling</b></div>
<div id="fix">

Consider wrapping the entire calculator logic in a `try-except` block to handle potential `KeyboardInterrupt` and other exceptions gracefully. This would provide a better user experience when exiting the program.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -29,6 +29,7 @@
 while True:
     try:
         # take input from the user
         choice = input("Enter choice(1/2/3/4): ")
     except (KeyboardInterrupt, EOFError):
         print("\nExiting calculator...")
         break

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/29#issuecomment-2535490008>#40d269</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged